lxd/network/openvswitch/ovn.go

```diff
@@ -1241,21 +1241,26 @@ func (o *OVN) LogicalSwitchPortAdd(switchName OVNSwitch, portName OVNSwitchPort,
 			args = append(args, string(opts.Parent), fmt.Sprint(opts.VLAN))
 		}
 
-		ipStr := make([]string, 0, len(opts.IPs))
-		for _, ip := range opts.IPs {
-			ipStr = append(ipStr, ip.String())
+		var addresses []string
+
+		if opts.MAC != nil {
+			addresses = append(addresses, opts.MAC.String())
 		}
 
-		var addresses string
-		if opts.MAC != nil && len(ipStr) > 0 {
-			addresses = opts.MAC.String() + " " + strings.Join(ipStr, " ")
-		} else if opts.MAC != nil && len(ipStr) <= 0 {
-			addresses = opts.MAC.String() + " dynamic"
-		} else {
-			addresses = "dynamic"
+		if len(opts.IPs) > 0 {
+			// If static IPs are requested then use these.
+			for _, ip := range opts.IPs {
+				addresses = append(addresses, ip.String())
+			}
+		} else if opts.DHCPv4OptsID != "" || opts.DHCPv6OptsID != "" {
+			// Or if either DHCPv4 or DHCPv6 is enabled then request dynamic IPs for both protocols.
+			// OVN currently doesn't allow us to request protocol specific dynamic IPs.
+			addresses = append(addresses, "dynamic")
 		}
 
-		args = append(args, "--", "lsp-set-addresses", string(portName), addresses)
+		if len(addresses) > 0 {
+			args = append(args, "--", "lsp-set-addresses", string(portName), strings.Join(addresses, " "))
```

Comment:
Is it intentional to join the addresses into an single long space-separated strings rather than append the slice?

Reply:
yep its intended

Reply:
ill double check to be sure, but that is what it was previously anyway and is still working as far as i can tell.

Reply:
although its not clear from the man page, it does seemingly make a difference as dhpv4 didnt work when separate args.